dotcom-rendering/src/components/ExpandableMarketingCard.tsx

```diff
@@ -0,0 +1,258 @@
+import type { SerializedStyles } from '@emotion/react';
+import { css } from '@emotion/react';
+import {
+	from,
+	headlineBold17,
+	headlineBold20,
+	neutral,
+	palette as sourcePalette,
+	space,
+	textSans15,
+	textSansBold12,
+	textSansBold14,
+} from '@guardian/source/foundations';
+import {
+	LinkButton,
+	SvgChevronDownSingle,
+	SvgCross,
+} from '@guardian/source/react-components';
+import type { Dispatch, SetStateAction } from 'react';
+import { getZIndex } from '../lib/getZIndex';
+import { palette } from '../palette';
+import { useConfig } from './ConfigContext';
+
+interface BannersLogoProps {
+	type: 'faded' | 'top' | 'bottom';
+	styles?: SerializedStyles;
+}
+
+const BannersLogo = ({ type, styles }: BannersLogoProps) => {
+	const { assetOrigin } = useConfig();
+	const src = `${assetOrigin}static/frontend/logos/red-blue-banner-${type}.svg`;
+	return <img src={src} alt="Guardian logo held up by hands" css={styles} />;
+};
+
+const fillBarStyles = css`
+	background-color: ${palette('--expandable-marketing-card-fill-background')};
+	width: 100%;
+	margin-top: -1px;
+	margin-bottom: -1px;
+
+	height: 6px;
+	${from.desktop} {
+		height: 4px;
+	}
+`;
+
+const containerStyles = css`
+	${getZIndex('expandableMarketingCardOverlay')}
+	position: sticky;
+	top: 0;
+	padding-bottom: ${space[5]}px;
+	margin-right: -1px; /* To align with rich link - if we move this feature to production, we should remove this and make rich link align with everything instead */
+`;
+
+const contentStyles = css`
+	position: relative;
+	display: flex;
+	flex-direction: column;
+	justify-content: space-between;
+	align-items: start;
+	gap: ${space[4]}px;
+	height: auto;
+	width: auto;
+	padding: ${space[2]}px;
+	border-radius: 0 0 ${space[2]}px ${space[2]}px;
+	background-color: ${palette('--expandable-marketing-card-background')};
+	color: ${neutral[100]};
+`;
+
+const summaryStyles = css`
+	display: flex;
+	flex-direction: column;
+	gap: ${space[3]}px;
+	z-index: 1;
+	width: 100%;
+`;
+
+const headingStyles = css`
+	display: flex;
+	gap: ${space[2]}px;
+	justify-content: space-between;
+
+	${headlineBold17};
+	${from.leftCol} {
+		${headlineBold20};
+	}
+`;
+
+const kickerStyles = css`
+	${textSans15};
+`;
+
+const arrowStyles = css`
+	flex-shrink: 0;
+	display: inline-flex;
+	align-items: center;
+	border: none;
+	vertical-align: middle;
+	justify-content: center;
+	padding: 0;
+	width: ${space[6]}px;
+	height: ${space[6]}px;
+	background-color: ${sourcePalette.neutral[100]};
```

Comment:
Should we add these Source colours to our DCR `palette`? I like the tightly-scoped and highly descriptive functions in the palette module, but I'm not aware if there's any rules around extending it, or in which cases we use the Source palette directly